src/server.js

```diff
@@ -256,12 +250,48 @@ function init(socket, client) {
 				client.names(data);
 			}
 		);
-		socket.join(client.id);
-		socket.emit("init", {
-			active: client.lastActiveChannel,
-			networks: client.networks,
-			token: client.config.token || null
+
+		socket.on("sign-out", (token) => {
+			delete client.config.sessions[token];
```

Comment:
Maybe just a tiny bit better code, but I'm not willing to rewrite and test it again.